minject/types.py

```diff
@@ -22,3 +23,17 @@ def __getitem__(self, key: K_contra) -> V_co:
 
     def __contains__(self, key: K_contra) -> bool:
         ...
+
+
+class _AsyncContext(Protocol):
+    """
+    Protocol for any object that can be used as an async context manager.
```

Comment:
can we update the documentation to specify what makes a `_AsyncContext` different from any async context managers? e.g. `__aenter__` must return `Self`